syntax = "proto2";

package logs.proto.tuningfork;

import "proto/tuningfork.proto";

message TuningForkLogEvent {
  // This may be unnecessary given there is a fingerprint in the Clearcut message
  optional string device_id = 1;
  // Can we get this from Clearcut, too?
  optional string apk_id = 2;
  optional .com.google.tuningfork.FidelityParams fidelityparams = 3;
  optional string experiment_id = 4;
  repeated TuningForkHistogram histograms = 5;
}

message TuningForkHistogram {
  // ID of the frame capture point
  required int32 instrument_id = 1;
  // Annotation extended by the developer
  required .com.google.tuningfork.Annotation annotation = 2;
  // Bucket counts.
  // The APK contains hard-coded bucket ranges for each instrument_id.
  repeated int32 counts = 3 [packed=true];
}
